sources/host-ctr/cmd/host-ctr/pid_namespaces.go

```diff
@@ -0,0 +1,184 @@
+}
+
+func addPidNsToManifest(pidNs int64) error {
+	fileLock := flock.New(lockfile)
+	lockCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	locked, err := fileLock.TryLockContext(lockCtx, 500*time.Millisecond)
+	if err != nil {
+		return errors.Wrap(err, "acquiring lock for writing to PID NS manifest failed with error")
+	}
+	if !locked {
+		return errors.New("failed to acquire lock for writing to PID NS manifest")
+	}
+	defer func() {
+		_ = fileLock.Unlock()
+	}()
+
+	// Add task PID NS to manifest
+	if err := os.MkdirAll(filepath.Dir(manifestPath), 0666); err != nil {
+		return errors.Wrap(err, "failed to create PID NS manifest directory")
+	}
+	manifest, err := os.OpenFile(manifestPath, os.O_RDWR|os.O_CREATE, 0664)
+	if err != nil {
+		return errors.Wrap(err, "failed to open PID NS manifest file")
+	}
+	defer manifest.Close()
+
+	manifestStat, err := manifest.Stat()
+	if err != nil {
+		return errors.Wrap(err, "failed to stat PID NS manifest file")
+	}
+	raw := make([]byte, manifestStat.Size())
+	if _, err = manifest.Read(raw); err != nil {
+		return errors.Wrap(err, "failed to read PID NS manifest file")
+	}
+	var (
+		tmpPidNsList     HostCtrPidNs
+		hostCtrPidNsList HostCtrPidNs
+	)
+	if len(raw) != 0 {
+		err = json.Unmarshal(raw, &tmpPidNsList)
+		if err != nil {
+			return errors.Wrap(err, "failed to unmarshal PID NS manifest")
+		}
+		for _, ele := range tmpPidNsList.PidNamespaces {
+			// If it already exists in the manifest list for whatever reason, we can just return without doing anything.
+			if ele == pidNs {
+				return nil
+			}
+		}
+	}
+	hostCtrPidNsList.PidNamespaces = append(tmpPidNsList.PidNamespaces, pidNs)
+	newData, err := json.MarshalIndent(hostCtrPidNsList, "", "  ")
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal PID NS list to JSON")
+	}
+	if err := manifest.Truncate(0); err != nil {
+		return errors.Wrap(err, "failed to truncate PID NS manifest")
+	}
+	if _, err := manifest.Seek(0, 0); err != nil {
+		return errors.Wrap(err, "failed to reset PID NS manifest I/O offset")
+	}
+	if _, err = manifest.Write(newData); err != nil {
+		return errors.Wrap(err, "failed to write to PID NS manifest")
+	}
+	return nil
+}
+
+func removePidNsFromManifest(pidNs int64) error {
+	fileLock := flock.New(lockfile)
+	lockCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	locked, err := fileLock.TryLockContext(lockCtx, 500*time.Millisecond)
+	if err != nil {
+		return errors.Wrap(err, "acquiring lock for writing to pid-ns manifest failed with error")
+	}
+	if !locked {
+		return errors.New("failed to acquire lock for writing to pid-ns manifest")
+	}
+	defer func() {
+		_ = fileLock.Unlock()
+	}()
+
+	// Remove task PID NS from manifest
+	if err := os.MkdirAll(filepath.Dir(manifestPath), 0666); err != nil {
+		return errors.Wrap(err, "failed to create PID NS manifest directory")
+	}
```

Comment:
You should create the manifest path with tmpfiles.d configurations, as we do with other `/etc` directories.